codec/src/main/java/io/netty/handler/codec/compression/CompressionCodecs.java

```diff
@@ -0,0 +1,98 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * Creates a new {@link CompressionEncoder} or a new {@link CompressionDecoder}
+ * with specified {@link CompressionFormat}.
+ */
+public final class CompressionCodecs {
+
+    /**
+     * Factory method for a new {@link CompressionEncoder}.
+     * <p/>
+     * Note, that {@link CompressionFormat#ZLIB_OR_NONE} format is not allowed for an encoder.
+     *
+     * @param format specified compression format.
+     *
+     * @throws UnsupportedCompressionFormatException if specified format is not supported.
+     */
+    public static <T extends CompressionEncoder> T newEncoder(CompressionFormat format) {
+        ObjectUtil.checkNotNull(format, "format");
+        switch (format) {
+            case ZLIB:
+                return (T) ZlibCodecFactory.newZlibEncoder(ZlibWrapper.ZLIB);
+            case GZIP:
+                return (T) ZlibCodecFactory.newZlibEncoder(ZlibWrapper.GZIP);
+            case DEFLATE:
+                return (T) ZlibCodecFactory.newZlibEncoder(ZlibWrapper.NONE);
+            case ZLIB_OR_NONE:
+                throw new IllegalArgumentException("format: " + format
+                        + " (" + ZlibWrapper.ZLIB_OR_NONE + " is not allowed for an encoder)");
+            case SNAPPY:
+                return (T) new SnappyFrameEncoder();
+            case BZIP2:
+                return (T) new Bzip2Encoder();
+            case LZF:
+                return (T) new LzfEncoder();
+            case LZ4:
+                return (T) new Lz4FrameEncoder();
+            case FASTLZ:
+                return (T) new FastLzFrameEncoder();
+            case LZMA:
+                return (T) new LzmaFrameEncoder();
+            default:
+                throw new UnsupportedCompressionFormatException(format);
+        }
+    }
+
+    /**
+     * Factory method for a new {@link CompressionDecoder}.
+     *
+     * @param format specified compression format.
+     *
+     * @throws UnsupportedCompressionFormatException if specified format is not supported.
+     */
+    public static <T extends CompressionEncoder> T newDecoder(CompressionFormat format) {
+        ObjectUtil.checkNotNull(format, "format");
+        switch (format) {
+            case ZLIB:
+                return (T) ZlibCodecFactory.newZlibDecoder(ZlibWrapper.ZLIB);
+            case GZIP:
+                return (T) ZlibCodecFactory.newZlibDecoder(ZlibWrapper.GZIP);
+            case DEFLATE:
+                return (T) ZlibCodecFactory.newZlibDecoder(ZlibWrapper.NONE);
+            case ZLIB_OR_NONE:
+                return (T) ZlibCodecFactory.newZlibDecoder(ZlibWrapper.ZLIB_OR_NONE);
+            case SNAPPY:
+                return (T) new SnappyFrameDecoder();
+            case BZIP2:
+                return (T) new Bzip2Decoder();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Impossible cast from io.netty.handler.codec.compression.Bzip2Decoder to io.netty.handler.codec.compression.CompressionEncoder in io.netty.handler.codec.compression.CompressionCodecs.newDecoder(CompressionFormat) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)